src/libYARP_sig/CMakeLists.txt

```diff
@@ -64,9 +69,12 @@ add_library(YARP::YARP_sig ALIAS YARP_sig)
 target_include_directories(YARP_sig PUBLIC $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
                                            $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>)
 
-target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES}
-                               PUBLIC YARP::YARP_conf
-                                      YARP::YARP_OS)
+if (YARP_HAS_JPEG)
+   target_compile_definitions(YARP_sig PRIVATE YARP_HAS_JPEG_C=1)
+   target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES} PUBLIC YARP::YARP_conf YARP::YARP_OS ${JPEG_LIBRARY})
+else()
+   target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES} PUBLIC YARP::YARP_conf YARP::YARP_OS)
+endif()
```

Comment:
Same here:
```cmake
target_link_libraries(YARP_sig PRIVATE ${ACE_LIBRARIES}
                               PUBLIC YARP::YARP_conf 
                                      YARP::YARP_OS)
if(YARP_HAS_JPEG)
  target_compile_definitions(YARP_sig PRIVATE YARP_HAS_JPEG_C=1)
  target_link_libraries(${JPEG_LIBRARY})
endif()

Reply:
Nitpick: according to CMake documentation, `JPEG_LIBRARY` is an internal variable and `JPEG_LIBRARIES` should be used instead https://cmake.org/cmake/help/v3.11/module/FindJPEG.html .